src/services/auth.service.ts

```diff
@@ -72,3 +73,89 @@ export async function verifySignupOtp(body: SignupOtpVerifyRequest): Promise<str
   });
   return token;
 }
+
+export async function handleLogin(body:{email:string, password:string, deviceId:string}):Promise<string>{
+
+  const {password, deviceId}=body;
+  /**pull it off separately, so I can change it to lowercase */
+  const email=body.email.toLowerCase()
+
+  const existingUserAuth=await UserAuthDb.findOne({email})
+
+  if(!existingUserAuth||!(await existingUserAuth.verifyPassword(password))){
+
+    throw  new BadRequestError('Invalid login details');
+
+  }else if (!existingUserAuth.recognisedDevices.includes(deviceId)){
+    /** generate, save and send a new verification otp for the user*/
+    const otp= generateOtp()
+
+    /**upsert-true...  */
+    await UserVerificationDb.updateOne({email}, {
+          email, otp, deviceId, type:OtpType.LOGIN,
+          expiresAt: new Date(Date.now() + (10 * 60 * 1000)),
+          userId:existingUserAuth.userId
+
+      },{upsert:true})
+
+    // send email function goes here
+
+    /**Break execution of the code*/
+    throw  new BadRequestError('Device not recognised, enter OTP sent to mail to verify')
+    /** create a verify otp endpoint to verify the otp*/
+  }
+
+  /**generate and save access_token for user*/
+  const accessToken= generateToken(
+    {
+      email,
+      deviceId,
+      type: JwtType.USER,
+      userId:existingUserAuth.userId
+    })
+
+  /**for first time login -> upsert-true*/
+  await  UserTokenDb.updateOne({email,userId:existingUserAuth.userId},{
+    email,
+    token:accessToken,
+    userId:existingUserAuth.userId
+  },{upsert:true})
```

Comment:
Upserting would be removing the record of other access tokens for other devices. Add `deviceId` to the userAuth token schema.